Scripts/Utilities/MaterialUtils.cs

```diff
@@ -105,6 +105,14 @@ public static Color RandomColor()
 			return new Color(r, g, b);
 		}
 
+		public static Color HueShift(Color color, float shift)
```

Comment:
This parameters are still regular parameters, so the color being passed in is still getting copy-constructed (because of its struct-ness).
C# does not allow ref this parameters...yet